src/middlewares/index.ts

```diff
@@ -0,0 +1,5 @@
+import { textParserMiddleware } from "./text_parser"
```

Comment:
Don't think this structure makes a lot of sense since it is only going to be one middleware.

Maybe `src/middelware.ts` is better.